### PR TITLE
First implementation

### PR DESCRIPTION
**Pull request by pbrisbin**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches: main
```
Comment (pbrisbin) [inline .github/workflows/ci.yml]: Probably don't need this. There's nothing to deploy.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+## Motivation
+
+Using out of date dependencies is a Software Supply Chain risk, as
+security-related patches may not be applied in a timely fashion.
+
+Dependabot and timely merges of its PRs is a solution to this. Requiring manual
+approval and merge of such PRs is typically busy work. With robust CI and QA
+processes, such PRs should be a "merge on Green" scenario for your team. Any
+process of an Engineer taking an action without tough should be automated.
```
Comment (pbrisbin) [inline README.md]:
```suggestion
process of an Engineer taking an action without thought should be automated.
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+## Usage
+
+```yaml
+on:
+  schedule:
+    # ...
+
+jobs:
+  mergabot:
```
Comment (pbrisbin) [inline README.md]:
```suggestion
  mergeabot:
```

File: action.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    required: true
+    default: "${{ github.repository }}"
+
+  github-token:
+    description: "Override GitHub token, if necessary"
+    required: true
+    default: "${{ github.token }}"
+
+  dry-run:
+    descriptoin: "Set to 1 to print, but not perform, any actions"
```
Comment (pbrisbin) [inline action.yml]:
```suggestion
    description: "Set to 1 to print, but not perform, any actions"
```

File: bin/automerge-prs [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+trap 'rm -rf "$tmp"' EXIT
+
+now_s=$(date +"%s")
+since_s=$((now_s - (quarantine_days * 24 * 60 * 60)))
+since=$(date -d "@$since_s" +"%Y-%m-%d")
+
+gh_pr list --limit 1000 --json number,title,author,updatedAt,reviewDecision |
+  jq --compact-output ".[] |
+    select(.author.login == \"dependabot\") |
+    select(.updatedAt <= \"$since\")
```
Comment (pbrisbin) [inline bin/automerge-prs]: Could (probably should) use `--search` instead. Will experiment.

File: bin/automerge-prs [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  printf '  Last updated: \e[35m%s\e[0m\n' "$updatedAt"
+  printf '  Current review decision: \e[35m%s\e[0m\n' "$reviewDecision"
+
+  case "$reviewDecision" in
+    CHANGES_REQUESTED)
+      printf '  \e[1;37m=>\e[0m \e[33mSkip\e[0m\n'
+      ;;
+    APPROVED)
+      printf '  \e[1;37m=>\e[0m \e[32mEnable auto-merge\e[0m\n'
+
```
Comment (pbrisbin) [inline bin/automerge-prs]:
```suggestion
```

File: README.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+## Inputs
+
+- `quarantine-days`: how many days since the last update on a PR before it
+  qualifies for auto-merge. Default is 5.
+
+  Note that **any update** (a review, a comment, etc) will reset the clock.
+
+- `strategy`: how to perform the [auto-]merge. Must be `merge`, `rebase`, or
+  `squash`. Default is `rebase`.
```
Comment (eborden) [inline README.md:44]: > Default is `rebase`

:man_cook: :kiss:

File: action.yml [hunk 8]
```diff
@@ -0,0 +1,46 @@
+name: mergeabot
+description: Auto-merge Dependabot PRs, only after quarantine days
+
+inputs:
+  quarantine-days:
+    description: |
+      How long PRs must have gone since their last update to qualify for
+      auto-merge. Default is 5, Set to -1 to disable
+    required: true
+    default: 5
```
Comment (eborden) [inline action.yml:10]: How do `default` and `required` play with each other? Does it make sense to have a default if I'm required to set this?

File: bin/automerge-prs [hunk 9]
```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+if [[ -z "${GITHUB_TOKEN:-""}" ]]; then
```
Comment (eborden) [inline bin/automerge-prs:4]: I wonder if https://github.com/charmbracelet provides a tool for making env var and param parsing nicer?

File: bin/automerge-prs [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+  printf '  Current review decision: \e[35m%s\e[0m\n' "$reviewDecision"
+
+  case "$reviewDecision" in
+    CHANGES_REQUESTED)
+      printf '  \e[1;37m=>\e[0m \e[33mSkip\e[0m\n'
+      ;;
+    APPROVED)
+      printf '  \e[1;37m=>\e[0m \e[32mEnable auto-merge\e[0m\n'
+      if ((!DRY_RUN)); then
+        gh_pr merge --auto "$number"
```
Comment (eborden) [inline bin/automerge-prs]: No strategy here?

Comment (pbrisbin) [reply bin/automerge-prs]: Oops. Will fix.

The risks of `DRY_RUN` on CI.

File: bin/automerge-prs (hunk 9 above)
Comment (pbrisbin) [reply bin/automerge-prs:4]: You're saying what I've got is not "nice"? :wink: 

They've got some good stuff, but I think it's more focused on interactive CLI tools, so I'd be surprised. I can check.

Comment (eborden) [reply bin/automerge-prs:4]: I spent too long trying to find a clip of the Derry Girls saying "it's just not as nice" and failed. :laughing:

File: action.yml (hunk 8 above)
Comment (pbrisbin) [reply action.yml:10]: > How do default and required play with each other? 

I think the best way to look at it is each in isolation:

- `required: true`: When you ask for it, you will either get a value or an error, you will never get an implicit `""` or `null`
- `default: x`: When they don't specify it, behave as if the user specified `x`

Conceptually, I want both of those behaviors individually, so this is how I tend to do it (and have seen it done).

>  Does it make sense to have a default if **I'm required to set** this?

As described above, that's not really how `required` works. _Someone_ is required to set it, including via `default`.

Comment (eborden) [reply action.yml:10]: What a strange setup. :+1:

Comment (pbrisbin) [reply action.yml:10]: Yeah, it's a bit of a cluster.

In the JS libs, when you access an input you can also pass,

```js
core.getInput("foo", { required: true })
```

And I have _no idea_ how that plays with the stuff from `action.yml`.

File: bin/automerge-prs (hunk 9 above)
Comment (pbrisbin) [reply bin/automerge-prs:4]: Haha.

Well, if this were actually useful outside of its invocation in the action I would probably add options, and then [reference myself](https://pbrisbin.com/posts/an-opinionated-guide-to-options-parsing-in-shell/) on how to do it nicest.

**Review by eborden — APPROVED** [review]
LGTM

**Pull request merged**